fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/HttpHeaderInjection.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.http.commons.api;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.util.Map;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.UriInfo;
+
+import org.fcrepo.kernel.api.models.FedoraResource;
+
+import org.slf4j.Logger;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.stereotype.Component;
+
+import com.google.common.collect.ListMultimap;
+
+/**
+ * Inject optional headers from external processes
+ *
+ * @author whikloj
+ * @since 2015-10-30
+ */
+@Component
+public class HttpHeaderInjection implements ApplicationContextAware {
```

Comment:
Prefer subclassing `ApplicationObjectSupport` to impl'ing `ApplicationContextAware`.